File: articles/algorithmic-thinking-vs-computational-research/c/reasoning_profiles.c
#include <stdio.h>

typedef struct {
  const char* name;
  double step;
  double decomp;
  double control;
  double test;
  double representation;
  double governance;
} Profile;

int main(void) {
  Profile profiles[] = {
    {"Recipe-like procedure", 86, 72, 70, 62, 42, 20},
    {"Classroom algorithm exercise", 90, 82, 84, 78, 62, 32},
    {"Search and ranking system", 72, 70, 76, 66, 78, 70},
    {"Public decision-support workflow", 68, 66, 64, 72, 80, 86},
    {"Scientific modeling workflow", 74, 78, 76, 82, 86, 74}
  };
  printf("name,algorithmic_score,computational_score,warning\n");
  for (int i = 0; i < 5; i++) {
    double algorithmic = 0.28*profiles[i].step + 0.24*profiles[i].decomp + 0.24*profiles[i].control + 0.24*profiles[i].test;
    double computational = 0.16*profiles[i].step + 0.14*profiles[i].decomp + 0.14*profiles[i].control + 0.14*profiles[i].test + 0.22*profiles[i].representation + 0.20*profiles[i].governance;
    printf("%s,%.3f,%.3f,Synthetic educational diagnostic only.\n", profiles[i].name, algorithmic, computational);
  }
  return 0;
}

File: articles/algorithmic-thinking-vs-computational-research/fortran/reasoning_profiles.f90
program reasoning_profiles
  implicit none
  integer :: i
  character(len=40), dimension(5) :: names
  real(8), dimension(5) :: step, decomp, control, test, representation, governance, algorithmic, computational

  names = (/ 'Recipe-like procedure                 ', 'Classroom algorithm exercise         ', 'Search and ranking system            ', 'Public decision-support workflow     ', 'Scientific modeling workflow         ' /)
  step = (/ 86.0d0, 90.0d0, 72.0d0, 68.0d0, 74.0d0 /)
  decomp = (/ 72.0d0, 82.0d0, 70.0d0, 66.0d0, 78.0d0 /)
  control = (/ 70.0d0, 84.0d0, 76.0d0, 64.0d0, 76.0d0 /)
  test = (/ 62.0d0, 78.0d0, 66.0d0, 72.0d0, 82.0d0 /)
  representation = (/ 42.0d0, 62.0d0, 78.0d0, 80.0d0, 86.0d0 /)
  governance = (/ 20.0d0, 32.0d0, 70.0d0, 86.0d0, 74.0d0 /)

  print '(A)', 'name algorithmic_score computational_score warning'
  do i = 1, 5
    algorithmic(i) = 0.28d0*step(i) + 0.24d0*decomp(i) + 0.24d0*control(i) + 0.24d0*test(i)
    computational(i) = 0.16d0*step(i) + 0.14d0*decomp(i) + 0.14d0*control(i) + 0.14d0*test(i) + 0.22d0*representation(i) + 0.20d0*governance(i)
    print '(A,1X,F8.3,1X,F8.3,1X,A)', trim(names(i)), algorithmic(i), computational(i), 'Synthetic educational diagnostic only.'
  end do
end program reasoning_profiles
